web/pages/notifications.tsx

```diff
@@ -69,60 +165,310 @@ export default function Notifications() {
+
+  useEffect(() => {
+    if (user) listenForPrivateUser(user.id, setPrivateUser)
+  }, [user])
+
+  useEffect(() => {
+    if (!privateUser) return
+    if (privateUser.notificationPreferences) {
+      setNotificationSettings(privateUser.notificationPreferences)
+    }
+    if (
+      privateUser.unsubscribedFromCommentEmails ||
+      privateUser.unsubscribedFromAnswerEmails
+    ) {
+      setEmailNotificationSettings('less')
+    }
+    if (
+      privateUser.unsubscribedFromResolutionEmails &&
+      privateUser.unsubscribedFromCommentEmails &&
+      privateUser.unsubscribedFromAnswerEmails
+    ) {
+      setEmailNotificationSettings('none')
+    }
+  }, [privateUser])
+
+  function changeEmailNotifications(newValue: notification_subscribe_types) {
+    if (!privateUser) return
+    setEmailNotificationSettings(newValue)
+    if (newValue === 'all') {
+      updatePrivateUser(privateUser.id, {
+        unsubscribedFromResolutionEmails: false,
+        unsubscribedFromCommentEmails: false,
+        unsubscribedFromAnswerEmails: false,
+      })
+    } else if (newValue === 'less') {
+      updatePrivateUser(privateUser.id, {
+        unsubscribedFromResolutionEmails: false,
+        unsubscribedFromCommentEmails: true,
+        unsubscribedFromAnswerEmails: true,
+      })
+    } else if (newValue === 'none') {
+      updatePrivateUser(privateUser.id, {
+        unsubscribedFromCommentEmails: true,
+        unsubscribedFromAnswerEmails: true,
+        unsubscribedFromResolutionEmails: true,
+      })
+    }
+  }
+
+  function changeInAppNotificationSettings(
+    newValue: notification_subscribe_types
+  ) {
+    if (!privateUser) return
+    setNotificationSettings(newValue)
+    updatePrivateUser(privateUser.id, {
+      notificationPreferences: newValue,
+    })
+  }
+
+  useEffect(() => {
+    if (privateUser && privateUser.notificationPreferences)
+      setNotificationSettings(privateUser.notificationPreferences)
+    else setNotificationSettings('all')
+  }, [privateUser])
+
+  if (!privateUser) {
+    return <LoadingIndicator spinnerClassName={'border-gray-500 h-4 w-4'} />
+  }
+
+  return (
+    <div className={'p-2'}>
+      <div>In App Notifications</div>
+      <ChoicesToggleGroup
+        currentChoice={notificationSettings}
+        choicesMap={{ All: 'all', Less: 'less', None: 'none' }}
+        setChoice={(choice) =>
+          changeInAppNotificationSettings(
+            choice as notification_subscribe_types
+          )
+        }
+        className={'col-span-4 w-24'}
+      />
+      <div className={'mt-4'}>Email Notifications</div>
+      <ChoicesToggleGroup
+        currentChoice={emailNotificationSettings}
+        choicesMap={{ All: 'all', Less: 'less', None: 'none' }}
+        setChoice={(choice) =>
+          changeEmailNotifications(choice as notification_subscribe_types)
+        }
+        className={'col-span-4 w-24'}
+      />
+    </div>
+  )
+}
+
+async function GetNotificationSummaryText(
```

Comment:
nit: lowercase function names if they are not componets

Reply:
thanks!